propel/resource_data_pool.go

```diff
@@ -287,31 +298,34 @@ func resourceDataPoolRead(ctx context.Context, d *schema.ResourceData, m any) di
 
 func resourceDataPoolUpdate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
 	c := m.(graphql.Client)
+	id := d.Id()
+	input := &pc.ModifyDataPoolInput{
+		IdOrUniqueName: &pc.IdOrUniqueName{Id: &id},
+	}
 
-	if d.HasChanges("unique_name", "description", "syncing") {
```

Comment:
Caught a bug over here. Every time there is a change in properties different from "syncing" it attempts to update it anyway. The `GetOk` does not seem to work when there are nested fields. 

To reproduce it you can just create an HTTP data pool, try to update its description and it fails saying that HTTP data pools cannot have syncing periods configured.

Reply:
So this fixes the bug? Do we have a way to add a test for that?

Reply:
Yes, it fixes it. I'll add a test!